### PR TITLE
Refactors ExtendedTaskList methods

### PR DESCRIPTION
**Pull request by gurkanindibay**
ExecuteTaskListIntoTupleDestWithParam and ExecuteTaskListIntoTupleDest are nearly the same. I parameterized and a made a reusable structure here

### CODE REVIEW

**Comment by tejeswarm** [comment]
@gurkanindibay did you verify `executionParams->paramListInfo = paramListInfo; ` is NULL for `ExecuteTaskListIntoTupleDest `under debugger?

File: src/backend/distributed/executor/adaptive_executor.c [hunk 1]
```diff
@@ -1015,14 +1019,14 @@ ExecuteTaskListOutsideTransaction(RowModifyLevel modLevel, List *taskList,
 
 
 /*
- * ExecuteTaskListIntoTupleDestWithParam is a proxy to ExecuteTaskListExtended() which uses
- * bind params from executor state, and with defaults for some of the arguments.
+ * Prepare execution parameters for a task list.It is used by the
+ * functions that execute task lists.
```
Comment (onurctirtir) [inline src/backend/distributed/executor/adaptive_executor.c:1023]: We start function comments with the name of the function. I'd also suggest renaming this to CreateDefaultExecutionParams to follow a naming convention that is similar to CreateBasicExecutionParams:
 
```suggestion
 * CreateDefaultExecutionParams returns execution params based on given (possibly null)
 * bind params (presumably from executor state) with defaults for some of the arguments.
```

File: src/backend/distributed/executor/adaptive_executor.c [hunk 2]
```diff
@@ -1015,14 +1019,14 @@ ExecuteTaskListOutsideTransaction(RowModifyLevel modLevel, List *taskList,
...
  */
-uint64
-ExecuteTaskListIntoTupleDestWithParam(RowModifyLevel modLevel, List *taskList,
-									  TupleDestination *tupleDest,
-									  bool expectResults,
-									  ParamListInfo paramListInfo)
+static ExecutionParams *
+GetExecutionParams(RowModifyLevel modLevel, List *taskList,
```
Comment (onurctirtir) [inline src/backend/distributed/executor/adaptive_executor.c:1026]:
```suggestion
CreateDefaultExecutionParams(RowModifyLevel modLevel, List *taskList,
```

File: src/backend/distributed/executor/adaptive_executor.c [hunk 3]
```diff
@@ -1049,17 +1069,8 @@ ExecuteTaskListIntoTupleDest(RowModifyLevel modLevel, List *taskList,
...
-		modLevel, taskList, targetPoolSize, localExecutionSupported
-		);
-
-	executionParams->xactProperties = DecideTransactionPropertiesForTaskList(
-		modLevel, taskList, false);
-	executionParams->expectResults = expectResults;
-	executionParams->tupleDestination = tupleDest;
-
+	ExecutionParams *executionParams = GetExecutionParams(modLevel, taskList, tupleDest,
+														  expectResults, NULL);
```
Comment (onurctirtir) [inline src/backend/distributed/executor/adaptive_executor.c:1073]: We usually assign default values to variables and then pass them down to the functions:

```suggestion
	ParamListInfo paramListInfo = NULL;
	ExecutionParams *executionParams = GetExecutionParams(modLevel, taskList, tupleDest,
														  expectResults, paramListInfo);
```

**Pull request merged**